### PR TITLE
Add replicaset resources by default when using the k8sattributesprocessor

### LINKED ISSUE #2823: Failed to watch *v1.ReplicaSet and Failed to list *v1.ReplicaSet
### Component(s)

_No response_

## Description

Given the OpenTelemetryCollector configuration:

```
  config: |
    receivers:
      otlp:
        protocols:
          grpc: {}
          http: {}
    processors:
      k8sattributes: {}
      batch: {}
    exporters:
      datadog:
        api:
          site: datadoghq.com
          key: ${env:DD_API_KEY}
    service:
      pipelines:
        metrics:
          receivers:
            - otlp
          processors:
            - k8sattributes
            - batch
          exporters:
            - datadog
```

The otelcol throws the error:
```
W0405 22:24:51.394408       1 reflector.go:539] k8s.io/client-go@v0.29.3/tools/cache/reflector.go:229: failed to list *v1.ReplicaSet: replicasets.apps is forbidden: User "system:serviceaccount:observability-system:opentelemetry-collector" cannot list resource "replicasets" in API group "apps" at the cluster scope
E0405 22:24:51.394454       1 reflector.go:147] k8s.io/client-go@v0.29.3/tools/cache/reflector.go:229: Failed to watch *v1.ReplicaSet: failed to list *v1.ReplicaSet: replicasets.apps is forbidden: User "system:serviceaccount:observability-system:opentelemetry-collector" cannot list resource "replicasets" in API group "apps" at the cluster scope
```
over and over again.

## Steps to Reproduce
Deploy the above provided configuration

## Expected Result
No errors

## Actual Result
Errors due to missing rbac permissions

### Kubernetes Version

1.28.3

### Operator version

0.97.1

### Collector version

0.97.0

### Environment information

_No response_

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by iblancasa**
Fixes #2823

### CODE REVIEW

File: internal/manifests/collector/parser/processor/processor_k8sattributes.go [hunk 1]
```diff
@@ -50,15 +50,24 @@ func (o *K8sAttributesParser) ParserName() string {
...
+		{
+			APIGroups: []string{"apps"},
+			Resources: []string{"replicasets"},
+			Verbs:     []string{"get", "watch", "list"},
+		},
+		{
+			APIGroups: []string{"extensions"},
+			Resources: []string{"replicasets"},
+			Verbs:     []string{"get", "watch", "list"},
+		},
```
Comment (swiatekm-sumo) [inline internal/manifests/collector/parser/processor/processor_k8sattributes.go:69]: Why do we need this? I don't think any of our supported K8s versions ship this version.

Comment (iblancasa) [reply internal/manifests/collector/parser/processor/processor_k8sattributes.go:69]: When it is not added, the processor doesn't work: #2823, https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32247

Comment (swiatekm-sumo) [reply internal/manifests/collector/parser/processor/processor_k8sattributes.go:69]: I mean `extensions/replicaset` specifically. That API group was removed in K8s 1.16: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#replicaset-v116

Comment (iblancasa) [reply internal/manifests/collector/parser/processor/processor_k8sattributes.go:69]: You're right. Let me test if it works without them 🤔

**Review by jaronoff97 — APPROVED** [review]

**Pull request merged**